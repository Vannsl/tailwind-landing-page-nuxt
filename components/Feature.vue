<template>
  <div class="feature" :class="{ left: isLeft }">
    <div class="feature-image">
      <slot name="svg" />
    </div>
    <div class="feature-content">
      <h3>{{ headline }}</h3>
      <p>
        {{ content }}
        <br />
        <br />Images from:
        <a href="https://undraw.co/">undraw.co</a>
      </p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Feature',
  props: {
    headline: {
      type: String,
      required: false
    },
    content: {
      type: String,
      required: false
    },
    isLeft: {
      type: Boolean,
      default: true,
      required: false
    }
  }
}
</script>

<style scoped>
.feature {
  @apply flex flex-wrap flex-col-reverse;
}

@screen sm {
  .feature {
    @apply flex-row;
  }

  .feature.left {
    @apply flex-row-reverse;
  }
}

.feature-image,
.feature-content {
  @apply w-full p-6 mt-6;
}

@screen sm {
  .feature-image,
  .feature-content {
    @apply w-1/2;
  }
}

.feature-content {
  @apply align-middle;
}

.feature-content h3 {
  @apply text-3xl text-gray-800 font-bold leading-none mb-3;
}

.feature-content p {
  @apply text-gray-600 mb-8;
}

.feature-content a {
  @apply text-orange-500 underline;
}
</style>
