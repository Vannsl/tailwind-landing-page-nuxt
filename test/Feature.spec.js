import { shallowMount } from '@vue/test-utils'
import Feature from '@/components/Feature'

const factory = () => {
  return shallowMount(Feature, {
    propsData: {
      headline: 'Lorem Ipsum',
      content: 'Lorem Ipsum dolor sit'
    }
  })
}

describe('Feature', () => {
  it('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the property headline', () => {
    const wrapper = factory()
    expect(wrapper.props('headline')).toBe('Lorem Ipsum')
  })

  it('has the property content', () => {
    const wrapper = factory()
    expect(wrapper.props('content')).toBe('Lorem Ipsum dolor sit')
  })
})
