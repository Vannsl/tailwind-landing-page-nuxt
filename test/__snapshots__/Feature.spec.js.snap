// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature renders properly 1`] = `
"<div class=\\"feature left\\"><div class=\\"feature-image\\"></div> <div class=\\"feature-content\\"><h3>Lorem Ipsum</h3> <p>
      Lorem Ipsum dolor sit
      <br> <br>Images from:
      <a href=\\"https://undraw.co/\\">undraw.co</a></p></div></div>"
`;
